lib/phoenix_html/form.ex

```diff
@@ -332,6 +332,10 @@ defmodule Phoenix.HTML.Form do
     do: "#{id}_#{field}"
   def input_id(name, field) when is_atom(name),
     do: "#{name}_#{field}"
+  def input_id(name, field, value) do
+    value_id = value |> String.downcase |> String.replace(~r/\W/, "_")
```

Comment:
Why do we need to downcase?

Reply:
We should probably set the Unicode flag in the regex: `~r/\W/u`.

Reply:
Downcasing is just to make it look consistent. "search_key_foo" looks better than "search_key_FoO". Gotcha on the unicode thing (coming from ruby I didn't know it's a thing)

Reply:
Let's remove `downcase` then. It is not necessary. If people want it downcase, then they can pass it explicitly using the input_id/3 function. :D

Reply:
Fair enough. Will update PR for tomorrow.